lib/house.rb

```diff
@@ -0,0 +1,49 @@
+class House
+  attr_reader :price, :address, :rooms, :details
+
+  def initialize(price, address)
+    @price = price[1..-1].to_i
+    @address = address
+    @rooms = []
+    @total_area = 0
+    @details = {"price" => @price,
+                "address" => @address
+               }
+  end
+
+  def add_room(room_object)
+    @rooms << room_object
+  end
+
+  def above_market_average?
+    if @price > 500000 then true else false end
+  end
+
+  def rooms_from_category(category)
+    rooms_in_category = []
+    @rooms.each do |room|
+      if room.category == category
+        rooms_in_category << room
+      end
+    end
+    rooms_in_category
```

Comment:
this works just fine, and is a good solution. `find_all` is a better tool in this case though, and can be done in a single line. 

```
def rooms_from_category(category)
   @rooms.find_all {|room| room.category == category}
end
```